src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/CompareSymbolsCorrectlyTests.cs

```diff
@@ -1141,5 +1141,98 @@ public void M2(IEnumerable<IPropertySymbol> source, IEnumerable<IPropertySymbol>
                 },
             }.RunAsync();
         }
+
+        [Fact, WorkItem(4956, "https://github.com/dotnet/roslyn-analyzers/issues/4956")]
+        public async Task RS1024_StringGetHashCode()
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+@"
+using System;
+
+class C
+{
+    void M()
+    {
+        ReadOnlySpan<char> testROS = default;
+        int hashCode = string.GetHashCode(testROS, StringComparison.OrdinalIgnoreCase);
+    }
+}"
+                        , SymbolEqualityComparerStubCSharp
+                    }
+                }
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task RS1024_GetHashCodeOnInt64()
+        {
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"
+using System;
+using Microsoft.CodeAnalysis;
+
+static class HashCodeHelper
+{
+    public static Int32 GetHashCode(Int64 x) => 0;
+    public static Int32 GetHashCode(ISymbol symbol) => [|symbol.GetHashCode()|];
+}
+
+public class C
+{
+    public int GetHashCode(Int64 obj)
+    {
+        return HashCodeHelper.GetHashCode(obj);
+    }
+
+    public int GetHashCode(ISymbol symbol)
+    {
+        return HashCodeHelper.GetHashCode(symbol);
+    }
+
+    public int GetHashCode(object o)
+    {
+        if (o is ISymbol symbol)
+        {
+            return [|HashCode.Combine(symbol)|];
```

Comment:
This test doesn't check the codefix behavior. If no codefix is offered, can you use the same source for FixedCode as well?

Reply:
Sure